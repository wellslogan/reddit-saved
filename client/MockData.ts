import { RedditPost, RedditComment } from '@models';

export const MockSubredditsList = [
  'AskReddit',
  'boston',
  'news',
  'Politics',
  'Patriots',
  'bestoflegaladvice',
];

export const MockSubmissions = [
  {
    kind: 't3',
    data: {
      author: 'test-user',
      created: 1542247776,
      domain: 'example.com',
      id: '1',
      score: 945,
      selftext: '**some** self text with [markdown](https://example.com)',
      selftext_html:
        '<b>some</b> self text with <a href="https://example.com">markdown',
      subreddit: 'testsubreddit',
      title: 'Test Post 1',
      url: 'https://www.reddit.com',
      permalink: 'https://www.reddit.com',
      num_comments: 23,
    },
  } as RedditPost,
  {
    kind: 't1',
    data: {
      author: 'test-user',
      body: '**some** text with [markdown](https://example.com)',
      body_html: '<b>some</b> text with <a href="https://example.com">markdown',
      created: 1542247776,
      id: '2',
      score: 101,
      link_title: 'https://www.reddit.com',
      link_url: 'https://www.reddit.com',
      link_permalink: 'https://www.reddit.com',
      link_author: 'test-user2',
      num_comments: 3089,
      subreddit: 'testsubreddit',
      permalink: 'https://www.reddit.com',
    },
  } as RedditComment,
  {
    kind: 't3',
    data: {
      author: 'test-user',
      created: 1542247776,
      domain: 'example.com',
      id: '3',
      score: 945,
      selftext: '**some** self text with [markdown](https://example.com)',
      selftext_html:
        '<b>some</b> self text with <a href="https://example.com">markdown',
      subreddit: 'testsubreddit',
      title: 'Test Post 1',
      url: 'https://www.reddit.com',
      permalink: 'https://www.reddit.com',
      num_comments: 23,
    },
  } as RedditPost,
];

export const MockIdentity = {
  name: 'test-user',
};
