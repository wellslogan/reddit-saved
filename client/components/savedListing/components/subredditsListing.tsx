import * as React from 'react';

type SubredditsListingProps = {
  subreddits: string[];
  onClickSubreddit: (subreddit: string) => void;
};

export const SubredditsListing: React.StatelessComponent<
  SubredditsListingProps
> = ({ subreddits, onClickSubreddit }) => (
  <ul>
    <li>filter by sub:</li>
    {subreddits.map(sub => (
      <li key={sub}>
        <button
          className="link-button"
          onClick={() => onClickSubreddit(sub)}
          data-testid={`filterSubBtn-${sub}`}
        >
          {sub}
        </button>
      </li>
    ))}
  </ul>
);

SubredditsListing.displayName = 'SubredditListing';
