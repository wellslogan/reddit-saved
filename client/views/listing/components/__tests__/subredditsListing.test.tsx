import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { SubredditsListing } from '../subredditsListing';
import { MockSubredditsList } from '@app/MockData';

describe('Component: SubredditsListing', () => {
  it('should render', () => {
    const subreddits = MockSubredditsList;
    const onClickSubreddit = jest.fn();
    const { container } = render(
      <SubredditsListing {...{ subreddits, onClickSubreddit }} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onClickSubreddit function onClick', () => {
    const subreddits = MockSubredditsList;
    const onClickSubreddit = jest.fn();
    const { getByTestId } = render(
      <SubredditsListing {...{ subreddits, onClickSubreddit }} />
    );

    fireEvent.click(getByTestId('filterSubBtn-boston'));
    expect(onClickSubreddit).toHaveBeenCalledWith('boston');
  });
});
