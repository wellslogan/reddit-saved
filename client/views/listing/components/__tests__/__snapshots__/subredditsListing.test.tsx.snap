// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SubredditsListing should render 1`] = `
<div>
  <ul>
    <li>
      filter by sub:
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-AskReddit"
      >
        AskReddit
      </button>
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-boston"
      >
        boston
      </button>
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-news"
      >
        news
      </button>
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-Politics"
      >
        Politics
      </button>
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-Patriots"
      >
        Patriots
      </button>
    </li>
    <li>
      <button
        class="link-button"
        data-testid="filterSubBtn-bestoflegaladvice"
      >
        bestoflegaladvice
      </button>
    </li>
  </ul>
</div>
`;
